stdlib/jdbc/src/main/java/org/ballerinax/jdbc/functions/CreateClient.java

```diff
@@ -41,8 +44,44 @@
 
     public static ObjectValue createClient(Strand strand, MapValue<String, Object> config, MapValue<String,
                 Object> globalPoolOptions) {
-        ObjectValue jdbcClient = SQLDatasourceUtils.createSQLDBClient(config, globalPoolOptions);
+        ObjectValue jdbcClient = createSQLDBClient(config, globalPoolOptions);
         jdbcClient.addNativeData(Constants.CONNECTOR_ID_KEY, UUID.randomUUID().toString());
         return jdbcClient;
     }
+
+    private static ObjectValue createSQLDBClient(MapValue<String, Object> clientEndpointConfig,
+            MapValue<String, Object> globalPoolOptions) {
+        String url = clientEndpointConfig.getStringValue(Constants.EndpointConfig.URL);
+        String username = clientEndpointConfig.getStringValue(Constants.EndpointConfig.USERNAME);
+        String password = clientEndpointConfig.getStringValue(Constants.EndpointConfig.PASSWORD);
+        MapValue<String, Object> dbOptions = (MapValue<String, Object>) clientEndpointConfig
+                .getMapValue(Constants.EndpointConfig.DB_OPTIONS);
+        MapValue<String, Object> poolOptions = (MapValue<String, Object>) clientEndpointConfig
+                .getMapValue(Constants.EndpointConfig.POOL_OPTIONS);
+        boolean userProvidedPoolOptionsNotPresent = poolOptions == null;
+        if (userProvidedPoolOptionsNotPresent) {
+            poolOptions = globalPoolOptions;
+        }
+        PoolOptionsWrapper poolOptionsWrapper = new PoolOptionsWrapper(poolOptions);
+        String dbType = url.split(":")[1].toUpperCase(Locale.getDefault());
+
+        SQLDatasource.SQLDatasourceParamsBuilder builder = new SQLDatasource.SQLDatasourceParamsBuilder(dbType);
+        SQLDatasource.SQLDatasourceParams sqlDatasourceParams = builder.withPoolOptions(poolOptionsWrapper)
+                .withJdbcUrl(url).withUsername(username).withPassword(password).withDbOptionsMap(dbOptions)
+                .withIsGlobalDatasource(userProvidedPoolOptionsNotPresent).build();
+
+        return createSQLClient(sqlDatasourceParams);
+    }
+
+    private static ObjectValue createSQLClient(SQLDatasource.SQLDatasourceParams sqlDatasourceParams) {
```

Comment:
Do we need a separate method for this

Reply:
Not really. We can merge with the caller.